### ISSUE TITLE
Chunk going missing

### ISSUE THREAD

**Issue by mbwilding** [fixer-m/snowflake-db-net-client#24]
Hey, sometimes the first chunk in a large query will not be included in the aggregated final result.
Have you experienced this? Potentially a SF issue but unsure. Running in a AWS .NET 6 lambda.

**Comment by fixer-m** [comment]
@mbwilding Hi! 
Well, that's weird. I tried different queries and always got the first chunk data. 
I would suggest to check total rows count in response metadata and rows count that aggregated result contains (they should be equal): 

```csharp
var result = await _snowflakeClient.QueryRawResponseAsync("select top 10000 * from SNOWFLAKE_SAMPLE_DATA.TPCH_SF1000.SUPPLIER;");

var chunksDownloadInfo = new ChunksDownloadInfo() { ChunkHeaders = result.ChunkHeaders, Chunks = result.Chunks, Qrmk = result.Qrmk };
var parsed = await ChunksDownloader.DownloadAndParseChunksAsync(chunksDownloadInfo);

var totalRowCountInChunks = result.Chunks.Sum(c => c.RowCount);
Assert.AreEqual(totalRowCountInChunks, parsed.Count); 
```
As well as you can check rows in a first chunk from raw response - does it actually contains first rows from query result? Or it contains rows from result with some offset? (That will prove that SF just didn't returned first chunk and returned second instead, for instance).

**Comment by mbwilding** [comment]
![Untitled](https://user-images.githubusercontent.com/8856912/180736454-94ceff2f-ffa6-40ee-8a9d-cc37d30a93f4.png)
Here is what I mean.

**Comment by mbwilding** [comment]
![1](https://user-images.githubusercontent.com/8856912/180746554-a54dc2af-3b70-449b-9f88-1fed0f3cc067.png)
![2](https://user-images.githubusercontent.com/8856912/180746562-eae3f876-a170-4889-8e0d-c7b090ab8094.png)

**Comment by fixer-m** [comment]
@mbwilding Hi! This issue was fixed in 0.4.1. Turns out that SF API sometimes returns rows in "chunks" and in a "rowset" at the same time.

**Comment by fixer-m** [comment]
Fixed in 0.4.1.
